src/ruby/puppetserver-lib/puppet/server/master.rb

```diff
@@ -77,6 +77,16 @@ def getModuleInfoForEnvironment(env)
     end
   end
 
+  def getAllModuleInfoForEnvironment()
+    all_envs = @env_loader.list
+    all_envs.collect do |env|
```

Comment:
Seems like we'd end up with an array of hashes that each have one element in them, the environment with associated modules.  Would it be better maybe to just have a single hash with the environment (key) to module info (values) as each entry in that hash?